flux/stdlib/influxdata/influxdb/rules_test.go

```diff
@@ -1939,6 +1939,513 @@ func TestPushDownWindowAggregateRule(t *testing.T) {
+		Name:    "BadMaxCol",
+		Context: haveCaps,
+		Rules:   rules,
+		Before:  simplePlan(window1m, "max", &badMaxSpec),
+		After: &plantest.PlanSpec{
+			Nodes: []plan.Node{
+				plan.CreatePhysicalNode("ReadGroup", &influxdb.ReadGroupPhysSpec{
+					ReadRangePhysSpec: readRange,
+					GroupMode:         flux.GroupModeBy,
+				}),
+				plan.CreatePhysicalNode("window", &window1m),
+				plan.CreatePhysicalNode("max", &badMaxSpec),
+			},
+			Edges: [][2]int{
+				{0, 1},
+				{1, 2},
+			},
+		},
+	})
+
+	// No match due to a collapsed node having a successor
+	// ReadRange -> group -> window -> min
+	//                             \-> min
+	tests = append(tests, plantest.RuleTestCase{
+		Name:    "CollapsedWithSuccessor1",
+		Context: haveCaps,
+		Rules:   rules,
+		Before: &plantest.PlanSpec{
+			Nodes: []plan.Node{
+				plan.CreateLogicalNode("ReadRange", &readRange),
+				plan.CreateLogicalNode("group", group(flux.GroupModeBy)),
+				plan.CreateLogicalNode("window", &window1m),
+				plan.CreateLogicalNode("min", minProcedureSpec()),
+				plan.CreateLogicalNode("min", minProcedureSpec()),
+			},
+			Edges: [][2]int{
+				{0, 1},
+				{1, 2},
+				{2, 3},
+				{2, 4},
+			},
+		},
+		After: &plantest.PlanSpec{
+			Nodes: []plan.Node{
+				plan.CreatePhysicalNode("ReadGroup", &influxdb.ReadGroupPhysSpec{
+					ReadRangePhysSpec: readRange,
+					GroupMode:         flux.GroupModeBy,
+				}),
+				plan.CreatePhysicalNode("window", &window1m),
+				plan.CreatePhysicalNode("min", minProcedureSpec()),
+				plan.CreatePhysicalNode("min", minProcedureSpec()),
+			},
+			Edges: [][2]int{
+				{0, 1},
+				{1, 2},
+				{1, 3},
+			},
+		},
+	})
+
+	// No match due to a collapsed node having a successor
+	// ReadRange -> group -> window -> min
+	//                   \-> window
+	tests = append(tests, plantest.RuleTestCase{
+		Name:    "CollapsedWithSuccessor2",
+		Context: haveCaps,
+		Rules:   rules,
+		Before: &plantest.PlanSpec{
+			Nodes: []plan.Node{
+				plan.CreateLogicalNode("ReadRange", &readRange),
+				plan.CreateLogicalNode("group", group(flux.GroupModeBy)),
+				plan.CreateLogicalNode("window", &window1m),
+				plan.CreateLogicalNode("min", minProcedureSpec()),
+				plan.CreateLogicalNode("window", &window2m),
+			},
+			Edges: [][2]int{
+				{0, 1},
+				{1, 2},
+				{2, 3},
+				{1, 4},
+			},
+		},
+		After: &plantest.PlanSpec{
+			Nodes: []plan.Node{
+				plan.CreatePhysicalNode("ReadGroup", &influxdb.ReadGroupPhysSpec{
+					ReadRangePhysSpec: readRange,
+					GroupMode:         flux.GroupModeBy,
+				}),
+				plan.CreatePhysicalNode("window", &window1m),
+				plan.CreatePhysicalNode("min", minProcedureSpec()),
+				plan.CreatePhysicalNode("window", &window2m),
+			},
+			Edges: [][2]int{
+				{0, 1},
+				{1, 2},
+				{0, 3},
+			},
+		},
+	})
+
+	// 2.x has a feature-off case, but we don't support feature flagging so we ignore it.
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.Name, func(t *testing.T) {
+			//t.Parallel()
```

Comment:
Does something break if we run in parallel?

Reply:
Nope, this was a hack for testing - oops.